sevctl/src/platform.rs

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use core::arch::x86_64::__cpuid_count;
+use std::convert::TryFrom;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Generation {
+    Naples,
+    Rome,
+}
+
+impl Generation {
+    pub fn ark(&self) -> &[u8] {
+        match self {
+            Generation::Naples => include_bytes!("naples/ark.cert"),
+            Generation::Rome => include_bytes!("rome/ark.cert"),
+        }
+    }
+
+    pub fn ask(&self) -> &[u8] {
+        match self {
+            Generation::Naples => include_bytes!("naples/ask.cert"),
+            Generation::Rome => include_bytes!("rome/ask.cert"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ModelNumberParsingFail(pub String);
+
+impl TryFrom<&str> for Generation {
+    type Error = ModelNumberParsingFail;
+
+    fn try_from(maybe_model: &str) -> Result<Self, Self::Error> {
+        let naples = regex::Regex::new(r"AMD EPYC 7..1[P]?").expect("Failed to compile regex");
+        let rome = regex::Regex::new(r"AMD EPYC 7..2[P]?").expect("Failed to compile regex");
+
```

Comment:
@npmccallum @haraldh - are you aware of any specific cpuid or combination of cpuid values that can be used to deduce the processor generation? I've been poking through the AMD developer manuals and haven't found anything conclusive, but then again, they aren't terribly loquacious in describing some of these fields.

This is my "solution" I hacked together based on how AMD names their product lines.

Reply:
@connorkuehl nope, haven't tried yet